internal/provider/resources/global_concurrency_limit.go

```diff
@@ -0,0 +1,344 @@
+				Default:     float64default.StaticFloat64(0),
+				Validators: []validator.Float64{
+					float64validator.AtLeast(0),
+				},
+			},
+		},
+	}
+}
+
+// Create creates a new global concurrency limit.
+func (r *GlobalConcurrencyLimitResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan GlobalConcurrencyLimitResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := r.client.GlobalConcurrencyLimits(plan.AccountID.ValueUUID(), plan.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.CreateClientErrorDiagnostic("Global Concurrency Limit", err))
+
+		return
+	}
+
+	globalConcurrencyLimit, err := client.Create(ctx, api.GlobalConcurrencyLimitCreate{
+		Name:               plan.Name.ValueString(),
+		Limit:              plan.Limit.ValueInt64(),
+		Active:             plan.Active.ValueBool(),
+		ActiveSlots:        plan.ActiveSlots.ValueInt64(),
+		SlotDecayPerSecond: plan.SlotDecayPerSecond.ValueFloat64(),
+	})
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Global Concurrency Limit", "create", err))
+
+		return
+	}
+
+	copyGlobalConcurrencyLimitToModel(globalConcurrencyLimit, &plan)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func copyGlobalConcurrencyLimitToModel(globalConcurrencyLimit *api.GlobalConcurrencyLimit, model *GlobalConcurrencyLimitResourceModel) diag.Diagnostics {
+	model.ID = types.StringValue(globalConcurrencyLimit.ID.String())
+	model.Created = customtypes.NewTimestampValue(*globalConcurrencyLimit.Created)
+	model.Updated = customtypes.NewTimestampValue(*globalConcurrencyLimit.Updated)
+	model.Name = types.StringValue(globalConcurrencyLimit.Name)
+	model.Limit = types.Int64Value(globalConcurrencyLimit.Limit)
+	model.Active = types.BoolValue(globalConcurrencyLimit.Active)
+	model.ActiveSlots = types.Int64Value(globalConcurrencyLimit.ActiveSlots)
+	model.SlotDecayPerSecond = types.Float64Value(globalConcurrencyLimit.SlotDecayPerSecond)
+
+	return nil
+}
+
+// Delete deletes a global concurrency limit.
+func (r *GlobalConcurrencyLimitResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state GlobalConcurrencyLimitResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := r.client.GlobalConcurrencyLimits(state.AccountID.ValueUUID(), state.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.CreateClientErrorDiagnostic("Global Concurrency Limit", err))
+
+		return
+	}
+
+	err = client.Delete(ctx, state.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Global Concurrency Limit", "delete", err))
+
+		return
+	}
+}
```

Comment:
```suggestion
	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
	if resp.Diagnostics.HasError() {
		return
	}
}
```

You'll need to update the state to reflect that the object is deleted